cpp/src/round/round.cu

```diff
@@ -0,0 +1,130 @@
+    cudf::round_option round,
+    rmm::mr::device_memory_resource* mr,
+    cudaStream_t stream)
+  {
+    auto result   = cudf::make_fixed_width_column(input.type(), input.size());
+    auto out_view = result->mutable_view();
+    auto const n  = static_cast<int32_t>(std::pow(10, std::abs(decimal_places)));
+
+    if (decimal_places == 0)
+      thrust::transform(rmm::exec_policy(stream)->on(stream),
+                        input.begin<T>(),
+                        input.end<T>(),
+                        out_view.begin<T>(),
+                        [] __device__(T e) -> T { return ::round(e); });
+    else if (decimal_places > 0)
+      thrust::transform(rmm::exec_policy(stream)->on(stream),
+                        input.begin<T>(),
+                        input.end<T>(),
+                        out_view.begin<T>(),
+                        [n] __device__(T e) -> T { return ::round(e * n) / n; });
+    else  // decimal_places < 0
+      thrust::transform(rmm::exec_policy(stream)->on(stream),
+                        input.begin<T>(),
+                        input.end<T>(),
+                        out_view.begin<T>(),
+                        [n] __device__(T e) -> T { return ::round(e / n) * n; });
+
+    return result;
+  }
+
+  template <typename T>
+  std::enable_if_t<std::is_integral<T>::value, std::unique_ptr<column>> operator()(
+    column_view const& input,
+    int32_t decimal_places,
+    cudf::round_option round,
+    rmm::mr::device_memory_resource* mr,
+    cudaStream_t stream)
+  {
+    auto result = cudf::make_fixed_width_column(input.type(), input.size());
+
+    // only need to handle the case where decimal_places is < zero
+    // integers by definition have no fractional part, so result of "rounding" is a no-op
+    if (decimal_places < 0) {
+      auto out_view = result->mutable_view();
+      auto const n  = static_cast<int64_t>(std::pow(10, -decimal_places));
+      auto const m  = n / 10;
+
+      thrust::transform(rmm::exec_policy(stream)->on(stream),
+                        input.begin<T>(),
+                        input.end<T>(),
+                        out_view.begin<T>(),
+                        [n, m] __device__(T e) -> T {
+                          auto const rounding_digit = static_cast<int64_t>(fabs((e / m) % 10));
+                          auto const digits         = e / n;
+                          auto const adjust         = e > 0 ? 1 : -1;
+                          return rounding_digit < 5 ? digits * n : (digits + adjust) * n;
+                        });
+    }
+
+    return result;
+  }
+};
+
+std::unique_ptr<column> round(column_view const& input,
+                              int32_t decimal_places,
+                              cudf::round_option round,
+                              rmm::mr::device_memory_resource* mr,
+                              cudaStream_t stream)
+{
+  CUDF_EXPECTS(round == round_option::HALF_UP, "HALF_EVEN currently not supported.");
```

Comment:
Why not just remove HALF_EVEN from the `round_option` enum until it is supported? :)

Reply:
Done.